src/inmanta/user_setup.py

```diff
@@ -0,0 +1,179 @@
+            "running this command locally?"
+        )
+    if value == "yes":
+        # make sure auth is on
+        if not server_config.server_enable_auth.get():
+            raise click.ClickException(
+                "Server authentication should be enabled before running the initial user setup. "
+                "The option auth in the server section should be enabled."
+            )
+
+        click.echo(f"{'Server authentication: ' : <50}{click.style('enabled', fg='green')}")
+
+        # make sure the method is set to database
+        if server_config.server_auth_method.get() != "database":
+            raise click.ClickException(
+                "The server authentication method should be set to database to continue. Make sure auth_method in the server "
+                "section is set to database"
+            )
+
+        click.echo(f"{'Server authentication method: ' : <50}{click.style('database', fg='green')}")
+
+        # make sure there is auth config that supports signing tokens
+        cfg = config.AuthJWTConfig.get_sign_config()
+        if cfg is None:
+            click.echo("Error: No signing config available in the configuration.")
+            click.echo("To use a new config, add the following to the configuration in /etc/inmanta/inmanta.d/auth.cfg:\n")
+            click.echo(generate_signing_config())
+            raise click.ClickException(
+                "Make sure signing configuration is added to the config. See the documentation for details."
+            )
+
+        click.echo(f"{'Authentication signing config: ' : <50}{click.style('found', fg='green')}")
+
+    else:
+        raise click.ClickException("The user setup was aborted as it was not executed locally on the orchestrator")
+
+
+async def get_connection_pool() -> asyncpg.pool.Pool:
+    """
+    connect to a connection pool. It is the responsibility of the caller to close the return connection pool
+    """
+    try:
+        database_host = server_config.db_host.get()
+        database_port = server_config.db_port.get()
+
+        database_username = server_config.db_username.get()
+        database_password = server_config.db_password.get()
+        connection_pool_min_size = server_config.db_connection_pool_min_size.get()
+        connection_pool_max_size = server_config.db_connection_pool_max_size.get()
+        connection_timeout = server_config.db_connection_timeout.get()
+        return await data.connect(
+            database_host,
+            database_port,
+            server_config.db_name.get(),
+            database_username,
+            database_password,
+            create_db_schema=False,
+            connection_pool_min_size=connection_pool_min_size,
+            connection_pool_max_size=connection_pool_max_size,
+            connection_timeout=connection_timeout,
+        )
+    except Exception:
+        raise ConnectionPoolException
```

Comment:
```suggestion
        raise ConnectionPoolException()
```

Reply:
I learned recently that both are valid, though I still prefer the instance over the class.